packages/plugin-ext/src/main/browser/view/tree-view-widget.tsx

```diff
@@ -272,6 +273,40 @@ export class PluginTree extends TreeImpl {
         }, update);
     }
 
+    override markAsChecked(node: Mutable<TreeNode>, checked: boolean): void {
+        function findParentsToChange(child: TreeNode, nodes: TreeNode[]): void {
+            if (child.parent) {
+                if (child.parent.checkboxInfo !== undefined && child.parent.checkboxInfo.checked !== checked) {
+                    if (!checked || child.parent.children.every(candidate => candidate.checkboxInfo?.checked || candidate.checkboxInfo === undefined || candidate === child)) {
+                        nodes.push(child.parent);
+                        findParentsToChange(child.parent, nodes);
+                    }
+                }
+            }
+        }
```

Comment:
Minor: Quite a lot of nesting in here:
```suggestion
            if (child.parent?.checkboxInfo !== undefined 
             && child.parent.checkboxInfo.checked !== checked
             && (!checked || child.parent.children.every(candidate => 
                 candidate.checkboxInfo?.checked 
                 || candidate.checkboxInfo === undefined 
                 || candidate === child
             ))) {
                 nodes.push(child.parent);
                 findParentsToChange(child.parent, nodes);
            }
        }
```